test/test.js

```diff
@@ -1508,3 +1508,47 @@ test('example with form-auto-content', (t) => {
     t.ok(/--.+\r\nContent-Disposition: form-data; name="myFile"; filename="LICENSE"\r\n.*/.test(res.payload))
   })
 })
+
+test('simulate invalid alter _lightMyRequest.isDone with end', (t) => {
+  t.plan(2)
+  let end = false
+  const dispatch = function (req, res) {
+    req.resume()
+    req._lightMyRequest.isDone = true
+    req.on('end', () => {
+      end = true
+    })
+  }
+
+  let replied = false
+  inject(dispatch, { method: 'GET', url: '/', simulate: { end: true } }, (notHandledErr, res) => {
+    replied = true
+  })
+
+  setTimeout(() => {
+    t.equal(end, true)
+    t.equal(replied, false)
+  }, 10)
```

Comment:
Please avoid `setTimeout`, maybe use `setImmediate` or `process.nextTick`?

Reply:
What's the difference between these options?

Reply:
@kibertoad  their behavior is different. 

@mcollina ```setImmediate``` or ```process.nextTick``` cannot pass the testes, they fire faster than the event.

Reply:
I tried some possibilities here, and the following snipper works:
```js
  process.nextTick(() => {
    setImmediate(() => {
      t.equal(end, true)
      t.equal(replied, false)
    })
  })
```

Not sure why the only `process.nextTick` not works, maybe an internal behavior on lib.

Reply:
@RafaelGSS combination of ```process.nextTick``` and ```setImmediate``` did the tick on my side too. I will change to your approach. It seems that even combination of two of them is slightly faster than ```setTimeout(fn, 10)```

Reply:
Agree

Reply:
I do not know their are ```t.pass()``` or ```t.fail()``` to use.
I will change the testing method and remove ```setTimeout```.